arroyo-types/src/api.rs

```diff
@@ -0,0 +1,74 @@
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+#[derive(Serialize, Deserialize, Clone, ToSchema, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct PipelinePost {
+    pub name: String,
+    pub definition: Option<String>,
+    pub udfs: Option<Vec<Udf>>,
+}
+
+#[derive(Serialize, Deserialize, Clone, ToSchema, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Pipeline {
+    pub id: String,
+    pub name: String,
+    pub definition: Option<String>,
+    pub udfs: Vec<Udf>,
+}
+
+#[derive(Serialize, Deserialize, Clone, ToSchema, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct JobPost {
+    pub pipeline_id: String,
+    pub running_desired: Option<bool>,
+    pub tasks: Option<u64>,
+}
+
+#[derive(Serialize, Deserialize, Clone, ToSchema, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Job {
+    pub id: String,
+    pub pipeline_id: String,
+    pub running_desired: bool,
+    pub state: String,
+    pub run_id: u64,
+    pub start_time: Option<u64>,
+    pub finish_time: Option<u64>,
+    pub tasks: Option<u64>,
+    pub failure_message: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, ToSchema, Debug, FromPrimitive)]
+#[serde(rename_all = "camelCase")]
+pub enum UdfLanguage {
+    Rust = 0,
```

Comment:
Do these need to be numeric? I believe that serde will support specifying enums via names: https://serde.rs/json.html.

In protobuf these are numeric because that's how protobuf enums work, but we don't need to do that for JSON.

Reply:
I've actually removed this for the first release of the api. Let's just only accept Rust udfs for now.